### PR TITLE
feat: updated exports for the logger

### PR DESCRIPTION
**Pull request by Keyrxng**
- adds a single `index.ts` export file
- adds `tsup.config.ts`
- improved naming convention for `PrettyLogsWithOk`
- removes sourcemap from build
- also exporting the util functions for cleaning logs which will be beneficial for when relying on these logs in tests

image

### CODE REVIEW

File: CHANGELOG.md
Comment (gentlementlegen) [inline CHANGELOG.md:1]: Are changes needed within the changelogs?

File: package.json [hunk 1]
```diff
@@ -4,44 +4,18 @@
...
-  "tsup": {
-    "entry": [
-      "src/logs.ts",
-      "src/pretty-logs.ts"
-    ],
-    "splitting": false,
-    "sourcemap": true,
-    "clean": true,
-    "dts": true,
-    "format": ["cjs", "esm"]
```
Comment (gentlementlegen) [inline package.json:17]: Did you try that esm and cjs both work by removing this?

**Comment by Keyrxng** [comment]
on it. How do we bump the package version, manually via package json or is there a certain flow to follow? @gentlementlegen

I saw you manually bumped it the commits so I done the same.

File: CHANGELOG.md
Comment (Keyrxng) [reply CHANGELOG.md:1]: My first time updating a changelog, things look ok?

Comment (gentlementlegen) [reply CHANGELOG.md:1]: Changelogs are automated by `release-please` so will be re-generated.

**Comment by gentlementlegen** [comment]
Release please will automatically generate the changelogs based on the conventional commits, bump the version accordingly, and publish the package once all of these changes are merge into the main branch, so no need to do it manually.

**Comment by Keyrxng** [comment]
> Release please will automatically generate the changelogs based on the conventional commits, bump the version accordingly, and publish the package once all of these changes are merge into the main branch, so no need to do it manually.

Right okay, so should I revert the manual bump I just committed or it doesn't matter because `release-please` will handle it anyway?

**Comment by gentlementlegen** [comment]
Correct, please revert the bump and the changelog changes.

**Comment by Keyrxng** [comment]
> Correct, please revert the bump and the changelog changes.

So unless I redo the pull and fix the conventional commits the changelog won't change I expect since I used `chore:` for each commit which seems to be ignored, if everything is automated that is. I will keep this in mind when committing to other npm packages going forward

**Comment by gentlementlegen** [comment]
@Keyrxng It is also valid to name your PR with a conventional commit name, it should be picked up.

I renamed, and yes moving forward it is always nicer to properly name the commits if they introduce fixes, features etc.

**Comment by Keyrxng** [comment]
That's a good feature and I know my commit habits are poor (but improving 😂) especially commit names, everything is a chore to me but I will do @gentlementlegen cheers

**Comment by Keyrxng** [comment]
@gentlementlegen happy to approve and merge this in so I can use it in `start-stop` please?

File: CHANGELOG.md
Comment (gentlementlegen) [reply CHANGELOG.md:1]: @Keyrxng please revert changes here and good to go.

**Comment by gentlementlegen** [comment]
@Keyrxng please fix [this comment](https://github.com/ubiquity/ubiquibot-logger/pull/23#discussion_r1669294246) and good to go.

**Comment by Keyrxng** [comment]
> @Keyrxng please fix [this comment](https://github.com/ubiquity/ubiquibot-logger/pull/23#discussion_r1669294246) and good to go.

The changes here are because of `prettier` and are purely markdown syntax changes (same output). I can replace `-` with `*` and add `CHANGELOG.md` to `.prettierignore`?

**Comment by gentlementlegen** [comment]
@Keyrxng It will be totally re-generated on next release so it doesn't really matter, but why not I guess.

**Review by gentlementlegen — APPROVED** [review]

**Pull request merged**